includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -679,7 +681,7 @@ public function delete_renewal_meta( $renewal_order ) {
 	 */
 	public function update_failing_payment_method( $subscription, $renewal_order ) {
 		$subscription->update_meta_data( '_stripe_customer_id', $renewal_order->get_meta( '_stripe_customer_id', true ) );
-		$subscription->update_meta_data( '_stripe_source_id', $renewal_order->get_meta( '_stripe_source_id', true ) );
+		$subscription->update_meta_data( '_stripe_source_id', WC_Stripe_Order_Helper::get_instance()->get_stripe_source( $renewal_order ) );
```

Comment:
I see there's an `update_stripe_source()` method in the order helper. Why are we not using that here?

Reply:
I just wanted to keep subscriptions out of this change. I left all the meta update/get methods from subscriptions unchanged. The reasons are to reduce the effort required to review this PR and leave room for us to decide later whether we want to create a subscription-specific class (if that makes sense at all).